### PR TITLE
Organize sections

### PR DESCRIPTION
**Pull request by DelazJ**
The idea is to have the following structure, basically moving the memory provider into the vector creation section (no idea whether it's desirable though)
image

### CODE REVIEW

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 1]
```diff
@@ -536,72 +536,74 @@ You can write vector layer files using :class:`QgsVectorFileWriter` class. It
 supports any other kind of vector file that OGR supports (GeoPackage, Shapefile, GeoJSON,
 KML and others).
```
Comment (DelazJ) [inline source/docs/pyqgis_developer_cookbook/vector.rst]: I suspect this introductory paragraph might need a revisit but can't do that.

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 2]
```diff
@@ -536,72 +536,74 @@ You can write vector layer files using :class:`QgsVectorFileWriter` class. It
 supports any other kind of vector file that OGR supports (GeoPackage, Shapefile, GeoJSON,
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/vector.rst]:
```suggestion
supports any other kind of vector format that OGR supports (GeoPackage, Shapefile, GeoJSON,
```

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 3]
```diff
@@ -536,72 +536,74 @@ You can write vector layer files using :class:`QgsVectorFileWriter` class. It
...
 KML and others).
 
-There are two possibilities how to export a vector layer:
+There are different possibilities how to export a vector layer.
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/vector.rst]:
```suggestion
There are several ways to export a vector layer.
```

**Comment by DelazJ** [comment]
@havatv Thanks. What I meant about the introduction is that it refers only to QgsVectorFileWriter while, afaics, memory provider does not use that so we should find something that may include its reference.

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 4]
```diff
@@ -529,79 +529,81 @@ create them easily. This is what you have to do:
...
 
-Writing Vector Layers
-=====================
+Creating Vector Layers
+======================
 
-You can write vector layer files using :class:`QgsVectorFileWriter` class. It
-supports any other kind of vector file that OGR supports (GeoPackage, Shapefile, GeoJSON,
-KML and others).
+You can create vector layer datasets using the :class:`QgsVectorFileWriter` class.
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/vector.rst]:
```suggestion
You can create vector layers using the :class:`QgsVectorLayer` class and export as vector datasets using the :class:`QgsVectorFileWriter` class.
```

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 5]
```diff
@@ -529,79 +529,81 @@ create them easily. This is what you have to do:
 
 .. index:: Vector layers; Creating
 
-Writing Vector Layers
-=====================
+Creating Vector Layers
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/vector.rst]:
```suggestion
Creating and Exporting Vector Layers
```

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 6]
```diff
@@ -529,79 +529,81 @@ create them easily. This is what you have to do:
...
+======================
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/vector.rst]:
```suggestion
===================================
```

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 7]
```diff
@@ -529,79 +529,81 @@ create them easily. This is what you have to do:
...
 
-You can write vector layer files using :class:`QgsVectorFileWriter` class. It
-supports any other kind of vector file that OGR supports (GeoPackage, Shapefile, GeoJSON,
-KML and others).
+You can create vector layer datasets using the :class:`QgsVectorFileWriter` class.
+It supports all the vector formats that OGR supports (GeoPackage, Shapefile,
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/vector.rst]:
```suggestion
The :class:`QgsVectorLayer` class can be used to create temporary, memory-based layers (`memory`) and
connect to OGR datasets (`ogr`), databases (`postgres`, `spatialite`, `mysql`) and more (`wfs`, `gpx`, `delimitedtext`, ...).
The :class:`QgsVectorFileWriter` class supports all OGR supported vector formats (GeoPackage,
Shapefile,
```

**Comment by DelazJ** [comment]
I revisit the sections, adding an introduction to the two involved classes. Texts are picked from their pyqgis doc but review is more than welcome.
btw we mentioned qgsvectorlayer while the first example is all about qgsvectorfilewriter.

File: source/docs/pyqgis_developer_cookbook/vector.rst [hunk 8]
```diff
@@ -534,91 +543,108 @@ create them easily. This is what you have to do:
...
-  you can set whether to export only selected features, pass further
-  driver-specific options for creation or tell the writer not to create
-  attributes --- look into the documentation for full syntax.
+The third parameter specifies output text encoding. Only some drivers need this
+for correct operation (Shapefile is one of those), but if you
+are not using international characters you do not have to care much about
+the encoding.
 
-* directly from features
+The fourth parameter that we left as ``None`` may specify destination CRS ---
```
Comment (havatv) [inline source/docs/pyqgis_developer_cookbook/vector.rst]:
```suggestion
The fourth parameter that we left as ``None`` may specify the destination CRS ---
```

**Pull request merged**